src/main/java/org/mskcc/cmo/CmoNewRequestPublisher.java

```diff
@@ -113,6 +200,39 @@ private String generateFailedRequestSamplesSummary() {
         return builder.toString();
     }
 
+    /**
+     * Calls LimsRest and returns the list of samples for a given request id.
+     * @param requestId
+     * @return Map
+     * @throws Exception
+     */
+    private List<String> getLimsRequestsByTimestamp(Date startTimestamp, Date endTimestamp) throws Exception {
+        Gson gson = new Gson();
+        String requestUrl = limsBaseUrl + limsRequestDeliveriesEndpoint
+                + String.valueOf(startTimestamp.getTime());
+        RestTemplate restTemplate = getRestTemplate();
+        HttpEntity<LinkedMultiValueMap<String, Object>> requestEntity = getRequestEntity();
+        ResponseEntity responseEntity = restTemplate.exchange(requestUrl,
+                HttpMethod.GET, requestEntity, Object.class);
+        List<Map> response = gson.fromJson(gson.toJson(responseEntity.getBody()), List.class);
+
+        // if endtimestamp is provided then use it to filter the response results
+        // otherwise simply return set of request ids from response as list of strings
+        List<String> requestIds = new ArrayList<>();
+        for (Map m : response) {
+            Double deliveryDate = (Double) m.get("deliveryDate");
+            Date deliveryDateTimestamp = new Date(Double.valueOf(deliveryDate).longValue());
+            if (endTimestamp != null && deliveryDateTimestamp.after(endTimestamp)) {
+                log.debug("Request delivery date not within specified range, it will be skipped: "
```

Comment:
Should this be log.info?

Reply:
I set it to debug so that it wouldnt clutter up the logs